internal/fxcontext/context.go

```diff
@@ -65,10 +68,17 @@ func (c *Context) Logger() ulog.Log {
 	return store.log
 }
 
-// WithLogger returns a new context with the input logger
-func (c *Context) WithLogger(logger ulog.Log) *Context {
+// WithContextAwareLogger returns a new context with a context-aware logger
+func (c *Context) WithContextAwareLogger(span opentracing.Span) *Context {
 	store := c.getStore()
-	store.log = logger
+	// Note that opentracing.Tracer does not expose the tracer id
+	// We only inject tracing information for jaeger.Tracer
+	if jSpanCtx, ok := span.Context().(jaeger.SpanContext); ok {
+		traceLogger := c.Logger().With(
```

Comment:
Hmm,  we should fix that. People should be able to add zap `Field`s directly and still get the same Sentry benefits. Particularly in a path like this (on the per-request loop) where we know the fields ahead of time, we **really** shouldn't be incurring the overhead of the `interface{}` conversion since we know it's expensive.

In other words, this should be something like:

```go
traceLogger := c.Logger().Typed().With(
  zap.StringField("traceID", traceID),
  zap.StringField("spanID", spanID),
)
```